src/core/nm-dispatcher.c

```diff
@@ -836,6 +844,56 @@ _dispatcher_call(NMDispatcherAction    action,
                                                 TRUE);
     start_at_msec       = nm_utils_get_monotonic_timestamp_msec();
 
+    dbus_unit = g_dbus_connection_call_sync(gl.dbus_connection,
+                                       SYSTEMD_DBUS_SERVICE,
+				       SYSTEMD_DBUS_PATH,
+				       SYSTEMD_DBUS_INTERFACE,
+				       "GetUnit",
+				       g_variant_new("(s)", "dbus.service"),
+				       G_VARIANT_TYPE("(o)"),
+				       G_DBUS_CALL_FLAGS_NONE,
+				       CALL_TIMEOUT,
+				       NULL,
+				       &sd_err);
```

Comment:
I think there should be no blocking calls to external services from NetworkManager.

Also, the overhead that every call to the dispatcher involves now two additional D-Bus calls doesn't seem acceptable to me.

Reply:
you will have to monitor (and cache) what is going on on systemd. That way the value is cached and readily accessible.

Also, we should support systems without systemd.